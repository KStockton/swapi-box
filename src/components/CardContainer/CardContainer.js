import React from 'react';
import Card from '../Card/Card'
import './CardContainer.scss'
const shortid = require('shortid')

const Container = ({category, topic}) => {

  let cards = category.map(item =>{
   return <Card {...item} key={shortid.generate()} topic={topic}/>
  })
  
  return (
    <section className="CardContainer-wrapper">
       {cards}
    </section>
  )
}

export default Container