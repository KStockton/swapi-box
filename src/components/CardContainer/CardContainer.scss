.CardContainer-wrapper {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin: 90px auto;
  height: 100%;
  background: #edfffd21;
  width: 100vw;
  border-radius: 8px;
  padding-top: 43px;
  background-color: black;
}

@media only screen and (max-width: 800px) {

  .CardContainer-wrapper {
    background: #edfffd21;
    height: 100%;
  }
}
