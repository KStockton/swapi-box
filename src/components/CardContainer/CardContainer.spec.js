import React from 'react';
import CardContainer from './CardContainer.js';
import { shallow } from 'enzyme';

const mockProp = [
  {
    homeworld: "Tatooine",
    language: "Galactic Basic",
    name: "Luke Skywalker",
    population: "200000",
    species: "Human"
  }, {
    homeworld: "Tatooine",
    language: "n/a",
    name: "C-3PO",
    population: "200000",
    species: "Droid"
  }
];

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer category={mockProp}/>);
  });

  it('should render correctly without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});