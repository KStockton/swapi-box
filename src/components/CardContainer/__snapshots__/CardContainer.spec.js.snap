// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render correctly without crashing 1`] = `
<section
  className="CardContainer-wrapper"
>
  <Card
    homeworld="Tatooine"
    key="YqPiqYwP6"
    language="Galactic Basic"
    name="Luke Skywalker"
    population="200000"
    species="Human"
  />
  <Card
    homeworld="Tatooine"
    key="DkWfC3jH1m"
    language="n/a"
    name="C-3PO"
    population="200000"
    species="Droid"
  />
</section>
`;
